pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -192,67 +163,256 @@ function parseJsonFiles(rootPath: string, defaultsPath: string, lockedPath: stri
-            // Ignore empty objects
-            regValue = null;
-          }
-        } finally {
-          nativeReg.closeKey(innerKey);
-        }
+      if (!registryKey) {
+        continue;
+      }
+      const defaultsKey = nativeReg.openKey(registryKey, DEFAULTS_HIVE_NAME, nativeReg.Access.READ);
+      const lockedKey = nativeReg.openKey(registryKey, LOCKED_HIVE_NAME, nativeReg.Access.READ);
+
+      try {
+        defaults = defaultsKey ? this.readRegistryUsingSchema(settings.defaultSettings, defaultsKey, [DEFAULTS_HIVE_NAME]) : {};
+        locked = lockedKey ? this.readRegistryUsingSchema(settings.defaultSettings, lockedKey, [LOCKED_HIVE_NAME]) : {};
+      } catch (err) {
+        console.error('Error reading deployment profile: ', err);
+      } finally {
+        nativeReg.closeKey(registryKey);
+        nativeReg.closeKey(defaultsKey);
+        nativeReg.closeKey(lockedKey);
+      }
+      // If we found something in the HKLM Defaults or Locked registry hive, don't look at the user's
+      // Alternatively, if the keys work, we could break, even if both hives are empty.
+      if (Object.keys(defaults).length || Object.keys(locked).length) {
+        break;
+      }
+    }
+
+    // Don't bother with the validator, because the registry-based reader validates as it reads.
+    return { defaults, locked };
+  }
+
+  protected fullRegistryPath(...pathParts: string[]): string {
+    return `${ this.keyName }\\${ this.registryPathProfile.join('\\') }\\${ pathParts.join('\\') }`;
+  }
+
+  /**
+   * Windows only. Read settings values from registry using schemaObj as a template.
+   * @param schemaObj the object used as a template for navigating registry.
+   * @param regKey the registry key obtained from nativeReg.openKey().
+   * @param pathParts the relative path to the current registry key, starting at 'Defaults' or 'Locked'
+   * @returns null, or the registry data as an object.
+   */
+  protected readRegistryUsingSchema(schemaObj: Record<string, any>, regKey: nativeReg.HKEY, pathParts: string[]): Record<string, any> {
+    const newObject: Record<string, any> = {};
+    const schemaKeys = Object.keys(schemaObj);
+    const commonKeys: Array<{schemaKey: string, registryKey: string}> = [];
+    const unknownKeys: string[] = [];
+    const userDefinedObjectKeys: Array<{schemaKey: string, registryKey: string}> = [];
+    let regValue: any;
+
+    // Drop the initial 'defaults' or 'locked' field
+    const pathPartsWithoutHiveType = pathParts.slice(1);
+
+    for (const originalKey of nativeReg.enumKeyNames(regKey)) {
+      const schemaKey = fixProfileKeyCase(originalKey, schemaKeys);
+      // "fixed case" means mapping existing keys in the registry (which typically supports case-insensitive lookups)
+      // to the actual case in the schema.
+
+      if (schemaKey === null) {
+        unknownKeys.push(originalKey);
+      } else if (haveUserDefinedObject(pathPartsWithoutHiveType.concat(schemaKey), isEquivalentIgnoreCase)) {
```

Comment:
Note that at this point `pathPartsWithoutHiveType.concat(schemaKey)` is all using the correct case anyway, so we could use case-sensitive comparison if we wanted to (because we fixed the case for everything).

Reply:
.... which lets me turn `haveUserDefinedObject` into a one-liner using core JS and lodash, and remove the `isEquivalentRespectCase` function